source/vibelog/web.d

```diff
@@ -112,25 +116,26 @@ private final class VibeLogWeb {
 		}
 	}
 
-	@path("/posts/:postname/post_comment")
-	void postComment(string name, string email, string homepage, string message, string _postname, HTTPServerRequest req)
+	@path("posts/:slug/post_comment")
+	void postComment(string name, string email, string homepage, string message, string _slug, HTTPServerRequest req)
 	{
-		auto post = m_ctrl.db.getPost(_postname);
-		enforce(post.commentsAllowed, "Posting comments is not allowed for this article.");
-
-		auto c = new Comment;
-		c.isPublic = true;
-		c.date = Clock.currTime().toUTC();
-		c.authorName = name;
-		c.authorMail = email;
-		c.authorHomepage = homepage;
-		c.authorIP = req.peer;
-		if (auto fip = "X-Forwarded-For" in req.headers) c.authorIP = *fip;
-		if (c.authorHomepage == "http://") c.authorHomepage = "";
-		c.content = message;
-		m_ctrl.db.addComment(post.id, c);
-
-		redirect(m_subPath ~ "posts/" ~ post.name);
+		auto post = m_ctrl.db.getPost(_slug);
+
+		if( post.commentsAllowed )
```

Comment:
I'd prefer the old behavior or outputting an explicit error message. `@errorDisplay!getPost` and the required additions to `getPost` would then be needed to display the message properly.